File: python/exercism/conditionals/raindrops.py
def convert(number):
    is_factor = lambda number, factor: number % factor == 0

    is_factor3 = is_factor(number, 3)
    is_factor5 = is_factor(number, 5)
    is_factor7 = is_factor(number, 7)

    sounds = "".join(
        [
            "Pling" if is_factor3 else "",
            "Plang" if is_factor5 else "",
            "Plong" if is_factor7 else "",
        ]
    )

    return sounds if len(sounds) > 0 else str(number)

File: ruby/exercism/conditionals/raindrops.rb
class Raindrops
  def self.convert(value)
    result = ''
    result += 'Pling' if factor3?(value)
    result += 'Plang' if factor5?(value)
    result += 'Plong' if factor7?(value)
    result += value.to_s if result.empty?

    result
  end

  def self.factor3?(value)
    value % 3 == 0
  end

  def self.factor5?(value)
    value % 5 == 0
  end

  def self.factor7?(value)
    value % 7 == 0
  end
end
